src/lib.rs

```diff
@@ -1,3 +1,114 @@
+//! # Floem
+//! Floem is cross-platform GUI framework for Rust 🦀. It aims to be extremely performant while providing world-class developer ergonomics.
+//!
+//! ## Views
+//! Floem models the UI using a tree of [Views](view::View) that are built once. `Views` react to state changes and events.
+//! Views ar self-contained components that can be composed together to create complex UIs.
+//!
+//! ## Ids and Id paths
+//! [Id](id::Id)s are unique identifiers for views. They're used to identify views in the view tree. These ids are assigned via the [AppContext](context::AppContext) and are unique across the entire application.
+//!
+//! ## Events
+//! Events are passed down from the window to the view tree. Each view can handle events and propagate them to its children.
+//! Events can be passed to a specific view using its [Id](id::Id) path. [Id](id::Id) paths are unique ways to quickly identify a view, its children and its parent in the tree.
+//!
+//! Floem is responsible for making a view active. for handling drag and drop mechanics
+//! It's a View's responsibility to handle events and decide whether they're handled.
+//!
+//! ## State management  
+//!
+//! You will probably want your view components to have some state. You should place any state that affects
+//! you view inside a signal so that you can react to updates and update the `View`. Signals are reactive values that can be read from and written to.
```

Comment:
I'd add a link to the excellent leptos_reactive documentation here and point people there (and also maybe to the leptos book) to learn more about reactivity / signals.

Reply:
done!